packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -312,7 +320,7 @@ export class Cluster extends Resource implements ICluster {
    * The auto scaling group that hosts the default capacity for this cluster.
    * This will be `undefined` if the default capacity is set to 0.
    */
-  public readonly defaultCapacity?: autoscaling.AutoScalingGroup;
+  public readonly defaultCapacity?: Nodegroup | autoscaling.AutoScalingGroup;
```

Comment:
Sadly this won’t work with jsii languages like Java so we don’t use unions.

Let’s just use defaultNodegroup for the nodegroup version.

Reply:
Sure! Let me fix it.